src/index.ts

```diff
@@ -0,0 +1,206 @@
+    taskRunner: runner,
+    log: defaultLogger,
+  } = fastify;
+
+  // save action when an item is created
+  const createItemTaskName = itemTaskManager.getCreateTaskName();
+  runner.setTaskPostHookHandler(createItemTaskName, async (item: Partial<Item>, actor, { log, handler }) => {
+    const member = actor as Member;
+    const extra = {memberId: actor.id, itemId: item.id};
+    const view = VIEWS.BUILDER.name;
+    const geo = null;
+    const action: Action = new BaseAction(actor.id, item.id, member.type, item.type, ACTION_TYPES.CREATE, view, geo, extra);
+    actionService.create(action, handler);
+  });
+
+  // save action when an item is deleted
+  const deleteItemTaskName = itemTaskManager.getDeleteTaskName();
+  runner.setTaskPostHookHandler(deleteItemTaskName, async (item: Partial<Item>, actor, { log, handler }) => {
+    const member = actor as Member;
+    const extra = {memberId: actor.id, itemId: item.id};
+    const view = VIEWS.BUILDER.name;
+    const geo = null;
+    const action: Action = new BaseAction(actor.id, null, member.type, item.type, ACTION_TYPES.DELETE, view, geo, extra);
+    actionService.create(action, handler);
+  });
+
+  // get all the actions matching the given `id`
+  fastify.get<{ Params: IdParam, Querystring: AnalyticsQueryParams }>(
+    '/item/:id/analytics',
+    { schema: getOne },
+    async ({ member, params: { id }, query: { requestedSampleSize, view }, log }, reply) => {
+      const itemId = id;
+      // get actions aplying the parameters (view and requestedSampleSize)
+      const t1 = new GetActionsTask(member, itemId, requestedSampleSize, view, actionService);
+      await runner.runSingle(t1);
+      const actions = t1.result;
+      console.log(actions)
+
+      // get item
+      const t2 = itemTaskManager.createGetTaskSequence(member, itemId);
+      const itemResponse = await runner.runSingleSequence(t2);
+      const item = itemResponse as Item;
+      console.log(item)
+
+      // get memberships of the item
+      const t3 = itemMembershipsTaskManager.createGetOfItemTaskSequence(member, itemId);
+      const membershipsResponse = await runner.runSingleSequence(t3);
+      const memberships = membershipsResponse as ItemMembership[];
+      console.log(memberships)
+
+      // get members of the item
+      const tasks = memberships.map((membership) => memberTaskManager.createGetTask(member, membership.memberId));
+      const membersResponse = await runner.runMultiple(tasks, log);
+      const members = membersResponse as Member[];
+      console.log(members)
+
+      const numActionsRetrieved = actions.length;
+      const metadata = {
+        numActionsRetrieved: numActionsRetrieved,
+        requestedSampleSize: parseInt(requestedSampleSize)
+      }
+
+      // generate responseData with actions, members, item, and metadata
+      const responseData: Analytics = new BaseAnalytics(actions, members, item, metadata);
```

Comment:
Just wondering, but won't we need children data? are we sending the children actions as well? If not we can create an issue for that (don't need to do it yourself).

Reply:
We are currently not sending children data. It is only retrieving the actions of the item itself. I create an issue.